tox.ini

```
@@ -0,0 +1,15 @@
+[tox]
+requires =
+    tox>=4.2
+env_list =
+    py311
+    py310
+    py38
+
+[testenv]
+commands =
+    pytest --cov betterproto
+poetry_dep_groups =
+    dev
+require_locked_deps = true
+require_poetry = true
```

I would suggest the following.

- Keep `isolated_build`
- Install the tox plugin as part of `tox`, then `poetry` is resolved hermetically

```suggestion
[tox]
requires =
    tox-poetry-installer[poetry]==1.0.0b1
    tox>=4.2
envlist =
    py3
env_list =
    py311
    py310
    py39
    py38
    py37
isolated_build = true
[testenv]
require_poetry = true
require_locked_deps = true
poetry_dep_groups =
    dev
commands =
    pytest --cov {posargs} betterproto
```

---

I don't have a problem with keeping `isolated_build`, but I'm not sure what the benefit is of specifying the requirement inside `tox.ini` rather than `pyproject.toml`. The `tox-poetry-installer` docs recommend against it, but I'm not sure what the behaviour difference would be. Having it in `pyproject.toml` has caught the fact that the project requires `python>=3.7` but `tox-poetry-installer` doesn't support 3.7, which having it in `tox.ini` seemed to not flag.

---

It appears that `isolated_build` is no longer an option, as the behaviour is now the default. https://tox.wiki/en/latest/upgrading.html#isolated-environment-on-by-default